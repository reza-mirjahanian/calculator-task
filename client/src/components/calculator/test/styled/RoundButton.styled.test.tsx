import React from 'react'
import renderer from 'react-test-renderer'
import 'jest-styled-components'

import {RoundBtn, RoundBtnOrange, RoundBtnWhite} from "../../styles/RoundButton.styled";


describe('RoundBtn.styled', () => {
    test('RoundBtn should look like a button', () => {
        const tree = renderer.create(<RoundBtn>7</RoundBtn>).toJSON();
        expect(tree).toMatchSnapshot();
        expect(tree).toHaveStyleRule('color', '#fff');
        expect(tree).toHaveStyleRule('display', 'flex');
        expect(tree).toHaveStyleRule('justify-content', 'center');
        expect(tree).toHaveStyleRule('width', '3rem');
        expect(tree).toHaveStyleRule('font-size', '1.6rem');
        expect(tree).toHaveStyleRule('opacity', '0.25', {
            modifier: ':hover',
        })
    })
    test('RoundBtn should look like smaller in mobile', () => {
        const tree = renderer.create(<RoundBtn>7</RoundBtn>).toJSON();
        expect(tree).toMatchSnapshot();

        let mediaQuery = 'only screen and (max-width:425px)';
        expect(tree).toHaveStyleRule('height', '2rem', {
            media: mediaQuery,
        });
        expect(tree).toHaveStyleRule('font-size', '1.2rem', {
            media: mediaQuery,
        });
        expect(tree).toHaveStyleRule('width', '2rem', {
            media: mediaQuery,
        });

    })


    test('RoundBtn should accept wide argument', () => {
        const tree = renderer.create(<RoundBtn wide="2">AC</RoundBtn>).toJSON();
        expect(tree).toMatchSnapshot();
        expect(tree).toHaveStyleRule('color', '#fff');
        expect(tree).toHaveStyleRule('display', 'flex');
        expect(tree).toHaveStyleRule('justify-content', 'center');
        expect(tree).toHaveStyleRule('width', '100%');
    })


    test('RoundBtnOrange should works', () => {
        const tree = renderer.create(<RoundBtnOrange>/</RoundBtnOrange>).toJSON();
        expect(tree).toMatchSnapshot();
        expect(tree).toHaveStyleRule('color', '#fff');
        expect(tree).toHaveStyleRule('background-color', '#ff9503');
        expect(tree).toHaveStyleRule('color', '#ff9503', {
            modifier: ':focus',
        })
    })

    test('RoundBtnWhite should works', () => {
        const tree = renderer.create(<RoundBtnWhite>/</RoundBtnWhite>).toJSON();
        expect(tree).toMatchSnapshot();
        expect(tree).toHaveStyleRule('color', '#1c1c1c');
        expect(tree).toHaveStyleRule('background-color', '#a5a5a5');
        expect(tree).toHaveStyleRule('background-color', '#fff', {
            modifier: ':focus',
        })
    })
});
