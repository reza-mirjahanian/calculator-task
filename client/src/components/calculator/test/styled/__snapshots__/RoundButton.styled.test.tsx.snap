// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RoundBtn.styled RoundBtn should accept wide argument 1`] = `
.c0 {
  position: relative;
  cursor: pointer;
  padding: 0;
  border: 0;
  outline: 0;
  width: 100%;
  height: 3rem;
  line-height: 3rem;
  font-size: 1.6rem;
  border-radius: 3rem;
  background-color: #333;
  color: #fff;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  -webkit-transition: color 0.15s ease,background-color 0.15s ease;
  transition: color 0.15s ease,background-color 0.15s ease;
  grid-row-start: 5;
  grid-column-start: 1;
  grid-row-end: 5;
  grid-column-end: span 2;
}

.c0:hover {
  opacity: 0.25;
}

@media only screen and (max-width:425px) {
  .c0 {
    height: 2rem;
    line-height: 2rem;
    font-size: 1.2rem;
    border-radius: 2rem;
    width: 100%;
  }
}

<button
  className="c0"
>
  AC
</button>
`;

exports[`RoundBtn.styled RoundBtn should look like a button 1`] = `
.c0 {
  position: relative;
  cursor: pointer;
  padding: 0;
  border: 0;
  outline: 0;
  width: 3rem;
  height: 3rem;
  line-height: 3rem;
  font-size: 1.6rem;
  border-radius: 3rem;
  background-color: #333;
  color: #fff;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  -webkit-transition: color 0.15s ease,background-color 0.15s ease;
  transition: color 0.15s ease,background-color 0.15s ease;
}

.c0:hover {
  opacity: 0.25;
}

@media only screen and (max-width:425px) {
  .c0 {
    height: 2rem;
    line-height: 2rem;
    font-size: 1.2rem;
    border-radius: 2rem;
    width: 2rem;
  }
}

<button
  className="c0"
>
  7
</button>
`;

exports[`RoundBtn.styled RoundBtn should look like smaller in mobile 1`] = `
.c0 {
  position: relative;
  cursor: pointer;
  padding: 0;
  border: 0;
  outline: 0;
  width: 3rem;
  height: 3rem;
  line-height: 3rem;
  font-size: 1.6rem;
  border-radius: 3rem;
  background-color: #333;
  color: #fff;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  -webkit-transition: color 0.15s ease,background-color 0.15s ease;
  transition: color 0.15s ease,background-color 0.15s ease;
}

.c0:hover {
  opacity: 0.25;
}

@media only screen and (max-width:425px) {
  .c0 {
    height: 2rem;
    line-height: 2rem;
    font-size: 1.2rem;
    border-radius: 2rem;
    width: 2rem;
  }
}

<button
  className="c0"
>
  7
</button>
`;

exports[`RoundBtn.styled RoundBtnOrange should works 1`] = `
.c0 {
  position: relative;
  cursor: pointer;
  padding: 0;
  border: 0;
  outline: 0;
  width: 3rem;
  height: 3rem;
  line-height: 3rem;
  font-size: 1.6rem;
  border-radius: 3rem;
  background-color: #333;
  color: #fff;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  -webkit-transition: color 0.15s ease,background-color 0.15s ease;
  transition: color 0.15s ease,background-color 0.15s ease;
}

.c0:hover {
  opacity: 0.25;
}

.c1 {
  background-color: #ff9503;
}

.c1:focus {
  background-color: #fff;
  color: #ff9503;
}

@media only screen and (max-width:425px) {
  .c0 {
    height: 2rem;
    line-height: 2rem;
    font-size: 1.2rem;
    border-radius: 2rem;
    width: 2rem;
  }
}

<button
  className="c0 c1"
>
  /
</button>
`;

exports[`RoundBtn.styled RoundBtnWhite should works 1`] = `
.c0 {
  position: relative;
  cursor: pointer;
  padding: 0;
  border: 0;
  outline: 0;
  width: 3rem;
  height: 3rem;
  line-height: 3rem;
  font-size: 1.6rem;
  border-radius: 3rem;
  background-color: #333;
  color: #fff;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  -webkit-transition: color 0.15s ease,background-color 0.15s ease;
  transition: color 0.15s ease,background-color 0.15s ease;
}

.c0:hover {
  opacity: 0.25;
}

.c1 {
  background-color: #a5a5a5;
  color: #1c1c1c;
}

.c1:focus {
  background-color: #fff;
  color: #1c1c1c;
}

@media only screen and (max-width:425px) {
  .c0 {
    height: 2rem;
    line-height: 2rem;
    font-size: 1.2rem;
    border-radius: 2rem;
    width: 2rem;
  }
}

<button
  className="c0 c1"
>
  /
</button>
`;
